src/library/Cardano/CoinSelection.hs

```diff
@@ -222,13 +192,13 @@ data CoinSelectionOptions i o e = CoinSelectionOptions
 --   to produce a 'CoinSelection'.
 --
 data CoinSelectionError e
-    = ErrUtxoBalanceInsufficient Word64 Word64
+    = ErrUtxoBalanceInsufficient Coin Coin
     -- ^ The UTxO balance was insufficient to cover the total payment amount.
     --
     -- Records the /UTxO balance/, as well as the /total value/ of the payment
     -- we tried to make.
     --
-    | ErrUtxoNotFragmentedEnough Word64 Word64
+    | ErrUtxoNotFragmentedEnough Integer Integer
```

Comment:
`Integer` is here better than Word64? Why not `Natural` if `Word64` is too restrictive ? :thinking:

Reply:
Well spotted. I've changed this (and other similar values) to use `Natural`.

Fixed in cc259b15b77499ce80a101608b44f55471b830b0.